components/accelerator/nvidia/peermem/watcher.go

```diff
@@ -0,0 +1,96 @@
+package peermem
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/leptonai/gpud/components"
+	"github.com/leptonai/gpud/pkg/common"
+	pkg_dmesg "github.com/leptonai/gpud/pkg/dmesg"
+	events_db "github.com/leptonai/gpud/pkg/events-db"
+	"github.com/leptonai/gpud/pkg/log"
+)
+
+type watcher struct {
+	ctx          context.Context
+	eventsStore  events_db.Store
+	dmesgWatcher pkg_dmesg.Watcher
+}
+
+func newWatcher(ctx context.Context, eventsStore events_db.Store) (*watcher, error) {
```

Comment:
LOL.. These watchers are same.. Later we can make the watcher as a helper in the dmesg package, renamed to WatcherPipeline?  Caller just needs to pass eventdb store and matcher.

Reply:
@eahydra Yeah, please https://github.com/leptonai/gpud/pull/406 review for this.